File: backend/src/main/java/com/apocalypse/thefall/dto/item/WeaponDto.java
package com.apocalypse.thefall.dto.item;

import com.apocalypse.thefall.entity.item.enums.ItemType;
import com.apocalypse.thefall.entity.item.enums.WeaponType;

import java.util.Map;
import java.util.Set;

public record WeaponDto(
        Long id,
        ItemType type,
        Map<String, String> names,
        Map<String, String> descriptions,
        double weight,
        int basePrice,
        String path,
        WeaponType weaponType,
        int requiredStrength,
        int requiredHands,
        Integer capacity,
        DamageTypeDto damageType,
        Set<WeaponModeDto> weaponModes,
        Set<AmmoDto> compatibleAmmo
) implements ItemDto {
}

File: backend/src/main/java/com/apocalypse/thefall/dto/item/WeaponDto.kt
package com.apocalypse.thefall.dto.item

import com.apocalypse.thefall.dto.character.stats.DataItemDto
import com.apocalypse.thefall.entity.item.enums.ItemType
import com.apocalypse.thefall.entity.item.enums.WeaponType

@JvmRecord
data class WeaponDto(
    override val id: Long?,
    override val type: ItemType?,
    override val names: MutableMap<String, String>?,
    override val descriptions: MutableMap<String, String>?,
    override val weight: Double?,
    override val basePrice: Int?,
    override val path: String?,
    val weaponType: WeaponType?,
    val requiredStrength: Int,
    val requiredHands: Int,
    val capacity: Int?,
    val damageType: DataItemDto?,
    val weaponModes: MutableSet<WeaponModeDto?>?,
    val compatibleAmmo: MutableSet<AmmoDto?>?
) : ItemDto 